Source/WebCore/platform/graphics/avfoundation/cf/MediaPlayerPrivateAVFoundationCF.cpp

```diff
@@ -1797,15 +1803,21 @@ void AVFWrapper::seekCompletedCallback(AVCFPlayerItemRef, Boolean finished, void
     }
 
     LOG(Media, "AVFWrapper::seekCompletedCallback(%p)", self);
-    self->m_owner->scheduleMainThreadNotification(MediaPlayerPrivateAVFoundation::Notification::SeekCompleted, static_cast<bool>(finished));
+    ASSERT(m_seekCompletion);
+    if (m_seekCompletion) {
+        (*m_seekCompletion)(finished ? MediaPlayerEnums::SeekResult::Completed : MediaPlayerEnums::SeekResult::Cancelled);
+        m_seekCompletion = nullptr;
```

Comment:
Shouldn't this clear first as `MediaPlayerPrivateAVFoundation::seekWithTolerance` does?